swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/Resource.kt

```diff
@@ -174,35 +174,31 @@ interface Identifiable : Named {
     return "${workConfiguration.notificationConfiguration.resourceUrl}/$resourceId"
   }
 
-  fun findMatchingAttribute(key: String, values: List<Any>): String? {
-    try {
-      val fieldValue = getProperty(key) as? String
-      if (propertyMatches(values, fieldValue)) {
-        return fieldValue
-      }
-    } catch (e: IllegalArgumentException) {
-      return null
+  fun matchResourceAttributes(propertyName: String, values: List<Any?>): Boolean {
+    return try {
+      (getProperty(propertyName) as? Any).matchesAny(values)
+    } catch (e: Exception) {
+      false
     }
-
-    return null
   }
 
-  private fun propertyMatches(values: List<Any>, fieldValue: String?): Boolean {
-    if (fieldValue == null) {
-      return false
+  private fun Any?.matchesAny(values: List<Any?>): Boolean {
+    if (this == null || this !is String) {
+      return values.contains(this)
```

Comment:
why not just `return false` here?

Reply:
We wouldn't want that to return false right. How about non strings or nullables?

Reply:
👍 makes sense